src/strict_fstream.hpp

```diff
@@ -16,42 +15,41 @@
  */
 namespace strict_fstream
 {
-/// Overload of error-reporting function, to enable use with VS (1)
-/// and POSIX signature found in MUSL on Alpine (2)
-/// Ref 1: http://stackoverflow.com/a/901316/717706
-/// Ref 2: http://stackoverflow.com/a/41956165
-static inline char* strerror_r_compat(int result, char* buffer, int err)
-{
-    if (result)
-    {
-        sprintf(buffer, "Unknown error: %d", err);
-    }
-    return buffer;
-}
 
-static inline char* strerror_r_compat(char* result, char*, int)
-{
-    return result;
-}
+// Workaround for broken musl implementation
+// Since musl insists that they are perfectly compatible, ironically enough,
+// they don't have a __musl__ or similar. But __NEED_size_t is defined in their
+// relevant header (and not in working implementations), so we can use that.
+#ifdef __NEED_size_t
+#define BROKEN_GNU_STRERROR_R
+#warning "Working around broken strerror_r() implementation in musl, remove when musl is fixed"
+#endif
 
+/// Overload of error-reporting function, to enable use with VS.
+/// Ref: http://stackoverflow.com/a/901316/717706
 static std::string strerror()
 {
-    std::string buff(80, '\0');
+    std::string buff(256, '\0');
 #ifdef _WIN32
-    if (strerror_s(&buff[0], buff.size(), errno) != 0)
-    {
+    if (strerror_s(buff.data(), buff.size(), errno) == 0) {
+        buff.resize(buff.find('\0'));
```

Comment:
If the error is too long (longer than the allocated buffer), the string is truncated and it does not end with \0. resize(npos) leads to segfault. This happened to me once when trying to open a non existing zip file with a very long filepath.

Let's change it maybe to sth like:
if(buff.find(..) != string::npos) buff.resize(...) else buff += 'WARNING: truncation';

Reply:
iirc I bumped the buffer size to fix that, but better to do it like you say.

Reply:
sounds good, both strategies complement each other.